core/src/api/api.rs

```diff
@@ -269,38 +270,40 @@ mod test {
 	#[test]
 	fn new_graph_state_with_capacity_sets_initial_hash_map_capacity() {
 		let capacity: usize = 5;
-		let new_state = TestGraphState::<TEST_CAPACITY>::with_capacity(capacity);
+		let new_state =
+			TestGraphState::<TEST_CAPACITY>::with_capacity(Environment::Mainnet, capacity);
 		assert!(new_state.user_map.capacity() >= capacity);
 	}
 
 	#[test]
 	fn new_graph_state_with_capacity_caps_initial_hash_map_capacity() {
-		let new_state = TestGraphState::<TEST_CAPACITY>::with_capacity(TEST_CAPACITY * 2);
+		let new_state =
+			TestGraphState::<TEST_CAPACITY>::with_capacity(Environment::Mainnet, TEST_CAPACITY * 2);
 		assert!(new_state.user_map.capacity() >= TEST_CAPACITY);
 	}
 
 	#[test]
 	fn graph_state_capacity() {
-		let state: TestGraphState = TestGraphState::new();
+		let state: TestGraphState = TestGraphState::new(Environment::Mainnet);
```

Comment:
lol, this scared me to reference "Mainnet" in a test suite, until I remembered that Mainnet config has nothing to do with communicating on-chain; it's just for constants & schema definition. Still scary to read when your eye first scans it, though 🤷🏻

Reply:
Yeah we would need some more tests with some non mainnet configs to make sure everything works as expected but will do that in another PR.